Supporting Axum 0.8

getting an error
```
error[E0277]: the trait bound `axum::extract::connect_info::IntoMakeServiceWithConnectInfo<axum::Router, std::net::SocketAddr>: axum_test::transport_layer::IntoTransportLayer` is not satisfied
   --> src/testing/request.rs:84:9
    |
83  |       let server = TestServer::new_with_config(
    |                    --------------------------- required by a bound introduced by this call
84  | /         boot.router
85  | |             .unwrap()
86  | |             .into_make_service_with_connect_info::<SocketAddr>(),
    | |________________________________________________________________^ the trait `axum_test::transport_layer::IntoTransportLayer` is not implemented for `axum::extract::connect_info::IntoMakeServiceWithConnectInfo<axum::Router, std::net::SocketAddr>`
    |
    = help: the following other types implement trait `axum_test::transport_layer::IntoTransportLayer`:
              axum::extract::connect_info::IntoMakeServiceWithConnectInfo<S, C>
              axum::routing::Router
              axum::routing::into_make_service::IntoMakeService<S>
              axum::serve::Serve<M, S>
              axum::serve::WithGracefulShutdown<M, S, F>
note: required by a bound in `axum_test::TestServer::new_with_config`
   --> /.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-test-16.4.1/src/test_server.rs:206:12
    |
204 |     pub fn new_with_config<A, C>(app: A, config: C) -> Result<Self>
    |            --------------- required by a bound in this associated function
205 |     where
206 |         A: IntoTransportLayer,
    |            ^^^^^^^^^^^^^^^^^^ required by this bound in `TestServer::new_with_config`
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,14 +32,15 @@ path = "examples/example-websocket-ping-pong/main.rs"
 required-features = ["ws"]
 
 [features]
-default = ["pretty-assertions"]
+# default = ["pretty-assertions"]
+default = ["all"]
 
-all = ["pretty-assertions", "yaml", "msgpack", "reqwest", "shuttle", "typed-routing", "ws"]
+all = ["pretty-assertions", "yaml", "msgpack", "reqwest", "typed-routing", "ws"]
 
 pretty-assertions = ["dep:pretty_assertions"]
 yaml = ["dep:serde_yaml"]
 msgpack = ["dep:rmp-serde"]
-shuttle = ["dep:shuttle-axum"]
+# shuttle = ["dep:shuttle-axum"]
 typed-routing = ["dep:axum-extra"]
 ws = ["axum/ws", "tokio/time", "dep:uuid", "dep:base64", "dep:tokio-tungstenite", "dep:futures-util"]
 reqwest = ["dep:reqwest"]
@@ -74,7 +75,7 @@ pretty_assertions = { version = "1.4", optional = true }
 serde_yaml = { version = "0.9", optional = true }
 
 # Shuttle
-shuttle-axum = { version = "0.49", optional = true }
+# shuttle-axum = { version = "0.49", optional = true }
 
 # MsgPack
 rmp-serde = { version = "1.3", optional = true }
@@ -94,8 +95,8 @@ reqwest = { version = "0.12", optional = true, features = ["cookies", "json", "s
 [dev-dependencies]
 axum = { version = "0.8", features = ["multipart", "tokio", "ws"] }
 axum-extra = { version = "0.10", features = ["cookie", "typed-routing", "query"] }
-axum-msgpack = "0.4"
-axum-yaml = "0.4"
+axum-msgpack = "0.5"
+axum-yaml = "0.5"
 futures-util = "0.3"
 local-ip-address = "0.6"
 rand = { version = "0.8", features = ["small_rng"] }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -86,7 +86,7 @@ axum-extra = { version = "0.10", features = ["typed-routing"], optional = true }
 uuid = { version = "1.11", optional = true, features = ["v4"]}
 base64 = { version = "0.22", optional = true }
 futures-util = { version = "0.3", optional = true }
-tokio-tungstenite = { version = "0.24", optional = true }
+tokio-tungstenite = { version = "0.26", optional = true }
 
 # Reqwest
 reqwest = { version = "0.12", optional = true, features = ["cookies", "json", "stream", "multipart", "rustls-tls"] }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,8 +32,7 @@ path = "examples/example-websocket-ping-pong/main.rs"
 required-features = ["ws"]
 
 [features]
-# default = ["pretty-assertions"]
-default = ["all"]
+default = ["pretty-assertions"]
 
 all = ["pretty-assertions", "yaml", "msgpack", "reqwest", "typed-routing", "ws"]
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "axum-test"
 authors = ["Joseph Lenton <josephlenton@gmail.com>"]
-version = "16.4.1"
+version = "17.0.0"
 rust-version = "1.75"
 edition = "2021"
 license = "MIT"
@@ -47,7 +47,7 @@ reqwest = ["dep:reqwest"]
 [dependencies]
 auto-future = "1.0"
 assert-json-diff = "2.0"
-axum = { version = "0.7.9", features = [] }
+axum = { version = "0.8.1", features = [] }
 anyhow = "1.0"
 bytes = "1.8"
 bytesize = "1.3.0"
@@ -80,7 +80,7 @@ shuttle-axum = { version = "0.49", optional = true }
 rmp-serde = { version = "1.3", optional = true }
 
 # Typed Routing
-axum-extra = { version = "0.9", features = ["typed-routing"], optional = true }
+axum-extra = { version = "0.10", features = ["typed-routing"], optional = true }
 
 # WebSockets
 uuid = { version = "1.11", optional = true, features = ["v4"]}
@@ -92,9 +92,8 @@ tokio-tungstenite = { version = "0.24", optional = true }
 reqwest = { version = "0.12", optional = true, features = ["cookies", "json", "stream", "multipart", "rustls-tls"] }
 
 [dev-dependencies]
-async-trait = "0.1"
-axum = { version = "0.7", features = ["multipart", "tokio", "ws"] }
-axum-extra = { version = "0.9", features = ["cookie", "typed-routing", "query"] }
+axum = { version = "0.8", features = ["multipart", "tokio", "ws"] }
+axum-extra = { version = "0.10", features = ["cookie", "typed-routing", "query"] }
 axum-msgpack = "0.4"
 axum-yaml = "0.4"
 futures-util = "0.3"

--- a/README.md
+++ b/README.md
@@ -66,11 +66,12 @@ The current version of Axum Test requires at least Axum v0.7.6.
 
 Here is a list of compatability with prior versions:
 
-| Axum Version            | Axum Test Version |
-|-------------------------|-------------------|
-| 0.7.6 to 0.7.9 (latest) | 16+ (latest)      |
-| 0.7.0 to 0.7.5          | 14, 15            |
-| 0.6                     | 13.4.1            |
+| Axum Version    | Axum Test Version |
+|-----------------|-------------------|
+| 0.8.0+ (latest) | 17+ (latest)      |
+| 0.7.6 to 0.7.9  | 16                |
+| 0.7.0 to 0.7.5  | 14, 15            |
+| 0.6             | 13.4.1            |
 
 ## Examples
 

--- a/README.md
+++ b/README.md
@@ -55,7 +55,6 @@ In both cases allowing you to run multiple servers, across multiple tests, all i
 | `pretty-assertions` | **on**            | Uses the [pretty assertions crate](https://crates.io/crates/pretty_assertions) on response `assert_*` methods.                    |
 | `yaml`              | _off_             | Enables support for sending, receiving, and asserting, [yaml content](https://yaml.org/).                                         |
 | `msgpack`           | _off_             | Enables support for sending, receiving, and asserting, [msgpack content](https://msgpack.org/index.html).                         |
-| `shuttle`           | _off_             | Enables support for building a `TestServer` an [`shuttle_axum::AxumService`](https://docs.rs/shuttle-axum/latest/shuttle_axum/struct.AxumService.html), for use with [Shuttle.rs](https://shuttle.rs). |
 | `typed-routing`     | _off_             | Enables support for using `TypedPath` in requests. See [axum-extra](https://crates.io/crates/axum-extra) for details.             |
 | `ws`                | _off_             | Enables WebSocket support. See [TestWebSocket](https://docs.rs/axum-test/latest/axum_test/struct.TestWebSocket.html) for details. |
 | `reqwest`           | _off_             | Enables the `TestServer` being able to create [Reqwest](https://docs.rs/axum-test/latest/axum_test/struct.TestWebSocket.html) requests for querying. |

--- a/examples/example-websocket-chat/main.rs
+++ b/examples/example-websocket-chat/main.rs
@@ -158,7 +158,7 @@ pub(crate) fn new_app() -> Router {
     let shared_state = Arc::new(RwLock::new(state));
 
     Router::new()
-        .route(&"/ws-chat/:name", get(route_get_websocket_chat))
+        .route(&"/ws-chat/{name}", get(route_get_websocket_chat))
         .with_state(shared_state)
 }
 

--- a/examples/example-websocket-chat/main.rs
+++ b/examples/example-websocket-chat/main.rs
@@ -104,7 +104,7 @@ async fn handle_chat(socket: WebSocket, username: String, state: SharedAppState)
                         .expect("Failed to build JSON message for sending");
 
                     sender
-                        .send(Message::Text(json_text))
+                        .send(Message::Text(json_text.into()))
                         .await
                         .expect("Failed to send message to socket");
                 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -350,7 +350,7 @@ mod integrated_test_typed_routing_and_query {
     use serde::Serialize;
 
     #[derive(TypedPath, Deserialize)]
-    #[typed_path("/path-query/:id")]
+    #[typed_path("/path-query/{id}")]
     struct TestingPathQuery {
         id: u32,
     }

--- a/src/test_request.rs
+++ b/src/test_request.rs
@@ -2288,7 +2288,6 @@ mod test_clear_cookies {
 mod test_add_header {
     use super::*;
     use crate::TestServer;
-    use axum::async_trait;
     use axum::extract::FromRequestParts;
     use axum::routing::get;
     use axum::Router;
@@ -2303,7 +2302,6 @@ mod test_add_header {
 
     struct TestHeader(Vec<u8>);
 
-    #[async_trait]
     impl<S: Sync> FromRequestParts<S> for TestHeader {
         type Rejection = (StatusCode, &'static str);
 
@@ -2349,7 +2347,6 @@ mod test_add_header {
 mod test_authorization {
     use super::*;
     use crate::TestServer;
-    use axum::async_trait;
     use axum::extract::FromRequestParts;
     use axum::routing::get;
     use axum::Router;
@@ -2360,7 +2357,6 @@ mod test_authorization {
     fn new_test_server() -> TestServer {
         struct TestHeader(String);
 
-        #[async_trait]
         impl<S: Sync> FromRequestParts<S> for TestHeader {
             type Rejection = (StatusCode, &'static str);
 
@@ -2409,7 +2405,6 @@ mod test_authorization {
 mod test_authorization_bearer {
     use super::*;
     use crate::TestServer;
-    use axum::async_trait;
     use axum::extract::FromRequestParts;
     use axum::routing::get;
     use axum::Router;
@@ -2420,7 +2415,6 @@ mod test_authorization_bearer {
     fn new_test_server() -> TestServer {
         struct TestHeader(String);
 
-        #[async_trait]
         impl<S: Sync> FromRequestParts<S> for TestHeader {
             type Rejection = (StatusCode, &'static str);
 
@@ -2469,7 +2463,6 @@ mod test_authorization_bearer {
 mod test_clear_headers {
     use super::*;
     use crate::TestServer;
-    use axum::async_trait;
     use axum::extract::FromRequestParts;
     use axum::routing::get;
     use axum::Router;
@@ -2484,7 +2477,6 @@ mod test_clear_headers {
 
     struct TestHeader(Vec<u8>);
 
-    #[async_trait]
     impl<S: Sync> FromRequestParts<S> for TestHeader {
         type Rejection = (StatusCode, &'static str);
 

--- a/src/test_server.rs
+++ b/src/test_server.rs
@@ -1688,8 +1688,7 @@ mod test_clear_cookies {
 #[cfg(test)]
 mod test_add_header {
     use super::*;
-
-    use axum::async_trait;
+    use crate::TestServer;
     use axum::extract::FromRequestParts;
     use axum::routing::get;
     use axum::Router;
@@ -1699,14 +1698,11 @@ mod test_add_header {
     use hyper::StatusCode;
     use std::marker::Sync;
 
-    use crate::TestServer;
-
     const TEST_HEADER_NAME: &'static str = &"test-header";
     const TEST_HEADER_CONTENT: &'static str = &"Test header content";
 
     struct TestHeader(Vec<u8>);
 
-    #[async_trait]
     impl<S: Sync> FromRequestParts<S> for TestHeader {
         type Rejection = (StatusCode, &'static str);
 
@@ -1749,8 +1745,7 @@ mod test_add_header {
 #[cfg(test)]
 mod test_clear_headers {
     use super::*;
-
-    use axum::async_trait;
+    use crate::TestServer;
     use axum::extract::FromRequestParts;
     use axum::routing::get;
     use axum::Router;
@@ -1760,21 +1755,18 @@ mod test_clear_headers {
     use hyper::StatusCode;
     use std::marker::Sync;
 
-    use crate::TestServer;
-
     const TEST_HEADER_NAME: &'static str = &"test-header";
     const TEST_HEADER_CONTENT: &'static str = &"Test header content";
 
     struct TestHeader(Vec<u8>);
 
-    #[async_trait]
     impl<S: Sync> FromRequestParts<S> for TestHeader {
         type Rejection = (StatusCode, &'static str);
 
         async fn from_request_parts(
             parts: &mut Parts,
             _state: &S,
-        ) -> Result<TestHeader, Self::Rejection> {
+        ) -> Result<Self, Self::Rejection> {
             parts
                 .headers
                 .get(HeaderName::from_static(TEST_HEADER_NAME))

--- a/src/test_server.rs
+++ b/src/test_server.rs
@@ -444,7 +444,7 @@ impl TestServer {
     /// use axum_test::TestServer;
     ///
     /// #[derive(TypedPath, Deserialize)]
-    /// #[typed_path("/users/:user_id")]
+    /// #[typed_path("/users/{user_id}")]
     /// struct UserPath {
     ///     pub user_id: u32,
     /// }
@@ -2374,7 +2374,7 @@ mod test_typed_get {
     use serde::Deserialize;
 
     #[derive(TypedPath, Deserialize)]
-    #[typed_path("/path/:id")]
+    #[typed_path("/path/{id}")]
     struct TestingPath {
         id: u32,
     }
@@ -2408,7 +2408,7 @@ mod test_typed_post {
     use serde::Deserialize;
 
     #[derive(TypedPath, Deserialize)]
-    #[typed_path("/path/:id")]
+    #[typed_path("/path/{id}")]
     struct TestingPath {
         id: u32,
     }
@@ -2442,7 +2442,7 @@ mod test_typed_patch {
     use serde::Deserialize;
 
     #[derive(TypedPath, Deserialize)]
-    #[typed_path("/path/:id")]
+    #[typed_path("/path/{id}")]
     struct TestingPath {
         id: u32,
     }
@@ -2476,7 +2476,7 @@ mod test_typed_put {
     use serde::Deserialize;
 
     #[derive(TypedPath, Deserialize)]
-    #[typed_path("/path/:id")]
+    #[typed_path("/path/{id}")]
     struct TestingPath {
         id: u32,
     }
@@ -2510,7 +2510,7 @@ mod test_typed_delete {
     use serde::Deserialize;
 
     #[derive(TypedPath, Deserialize)]
-    #[typed_path("/path/:id")]
+    #[typed_path("/path/{id}")]
     struct TestingPath {
         id: u32,
     }
@@ -2544,7 +2544,7 @@ mod test_typed_method {
     use serde::Deserialize;
 
     #[derive(TypedPath, Deserialize)]
-    #[typed_path("/path/:id")]
+    #[typed_path("/path/{id}")]
     struct TestingPath {
         id: u32,
     }

--- a/src/test_web_socket.rs
+++ b/src/test_web_socket.rs
@@ -17,6 +17,7 @@ use crate::WsMessage;
 
 #[cfg(feature = "pretty-assertions")]
 use pretty_assertions::assert_eq;
+use tokio_tungstenite::tungstenite::Utf8Bytes;
 
 pub struct TestWebSocket {
     stream: WebSocketStream<TokioIo<Upgraded>>,
@@ -42,7 +43,8 @@ impl TestWebSocket {
         T: Display,
     {
         let text = format!("{}", raw_text);
-        self.send_message(WsMessage::Text(text)).await;
+        let text_bytes: Utf8Bytes = text.try_into().expect("Text contains non-Utf8 chars");
+        self.send_message(WsMessage::Text(text_bytes)).await;
     }
 
     pub async fn send_json<J>(&mut self, body: &J)
@@ -51,8 +53,9 @@ impl TestWebSocket {
     {
         let raw_json =
             ::serde_json::to_string(body).expect("It should serialize the content into Json");
+        let json_bytes: Utf8Bytes = raw_json.try_into().expect("Json contains non-Utf8 chars");
 
-        self.send_message(WsMessage::Text(raw_json)).await;
+        self.send_message(WsMessage::Text(json_bytes)).await;
     }
 
     #[cfg(feature = "yaml")]
@@ -62,8 +65,9 @@ impl TestWebSocket {
     {
         let raw_yaml =
             ::serde_yaml::to_string(body).expect("It should serialize the content into Yaml");
+        let yaml_bytes: Utf8Bytes = raw_yaml.try_into().expect("Yaml contains non-Utf8 chars");
 
-        self.send_message(WsMessage::Text(raw_yaml)).await;
+        self.send_message(WsMessage::Text(yaml_bytes)).await;
     }
 
     #[cfg(feature = "msgpack")]
@@ -74,7 +78,8 @@ impl TestWebSocket {
         let body_bytes =
             ::rmp_serde::to_vec(body).expect("It should serialize the content into MsgPack");
 
-        self.send_message(WsMessage::Binary(body_bytes)).await;
+        self.send_message(WsMessage::Binary(body_bytes.into()))
+            .await;
     }
 
     pub async fn send_message(&mut self, message: WsMessage) {
@@ -203,12 +208,18 @@ impl TestWebSocket {
 
 fn message_to_text(message: WsMessage) -> Result<String> {
     let text = match message {
-        WsMessage::Text(text) => text,
-        WsMessage::Binary(data) => String::from_utf8(data).map_err(|err| err.utf8_error())?,
-        WsMessage::Ping(data) => String::from_utf8(data).map_err(|err| err.utf8_error())?,
-        WsMessage::Pong(data) => String::from_utf8(data).map_err(|err| err.utf8_error())?,
+        WsMessage::Text(text) => text.to_string(),
+        WsMessage::Binary(data) => {
+            String::from_utf8(data.to_vec()).map_err(|err| err.utf8_error())?
+        }
+        WsMessage::Ping(data) => {
+            String::from_utf8(data.to_vec()).map_err(|err| err.utf8_error())?
+        }
+        WsMessage::Pong(data) => {
+            String::from_utf8(data.to_vec()).map_err(|err| err.utf8_error())?
+        }
         WsMessage::Close(None) => String::new(),
-        WsMessage::Close(Some(frame)) => frame.reason.into_owned(),
+        WsMessage::Close(Some(frame)) => frame.reason.to_string(),
         WsMessage::Frame(_) => {
             return Err(anyhow!(
                 "Unexpected Frame, did not expect Frame message whilst reading"
@@ -221,12 +232,12 @@ fn message_to_text(message: WsMessage) -> Result<String> {
 
 fn message_to_bytes(message: WsMessage) -> Result<Bytes> {
     let bytes = match message {
-        WsMessage::Text(string) => string.into_bytes().into(),
+        WsMessage::Text(string) => string.into(),
         WsMessage::Binary(data) => data.into(),
         WsMessage::Ping(data) => data.into(),
         WsMessage::Pong(data) => data.into(),
         WsMessage::Close(None) => Bytes::new(),
-        WsMessage::Close(Some(frame)) => frame.reason.into_owned().into_bytes().into(),
+        WsMessage::Close(Some(frame)) => frame.reason.into(),
         WsMessage::Frame(_) => {
             return Err(anyhow!(
                 "Unexpected Frame, did not expect Frame message whilst reading"
@@ -254,8 +265,8 @@ mod test_assert_receive_text {
                 while let Some(maybe_message) = socket.recv().await {
                     let message_text = maybe_message.unwrap().into_text().unwrap();
 
-                    let encoded_text = format!("Text: {message_text}");
-                    let encoded_data = format!("Binary: {message_text}").into_bytes();
+                    let encoded_text = format!("Text: {message_text}").try_into().unwrap();
+                    let encoded_data = format!("Binary: {message_text}").into_bytes().into();
 
                     socket.send(Message::Text(encoded_text)).await.unwrap();
                     socket.send(Message::Binary(encoded_data)).await.unwrap();
@@ -354,7 +365,7 @@ mod test_assert_receive_text_contains {
             async fn handle_ping_pong(mut socket: WebSocket) {
                 while let Some(maybe_message) = socket.recv().await {
                     let message_text = maybe_message.unwrap().into_text().unwrap();
-                    let encoded_text = format!("Text: {message_text}");
+                    let encoded_text = format!("Text: {message_text}").try_into().unwrap();
 
                     socket.send(Message::Text(encoded_text)).await.unwrap();
                 }
@@ -437,12 +448,15 @@ mod test_assert_receive_json {
                         "format": "text",
                         "message": decoded
                     }))
+                    .unwrap()
+                    .try_into()
                     .unwrap();
                     let encoded_data = serde_json::to_vec(&json!({
                         "format": "binary",
                         "message": decoded
                     }))
-                    .unwrap();
+                    .unwrap()
+                    .into();
 
                     socket.send(Message::Text(encoded_text)).await.unwrap();
                     socket.send(Message::Binary(encoded_data)).await.unwrap();
@@ -522,13 +536,15 @@ mod test_assert_receive_yaml {
                         "format": "text",
                         "message": decoded
                     }))
+                    .unwrap()
+                    .try_into()
                     .unwrap();
                     let encoded_data = serde_yaml::to_string(&json!({
                         "format": "binary",
                         "message": decoded
                     }))
                     .unwrap()
-                    .into_bytes();
+                    .into();
 
                     socket.send(Message::Text(encoded_text)).await.unwrap();
                     socket.send(Message::Binary(encoded_data)).await.unwrap();
@@ -608,7 +624,8 @@ mod test_assert_receive_msgpack {
                         "format": "binary",
                         "message": decoded
                     }))
-                    .unwrap();
+                    .unwrap()
+                    .into();
 
                     socket.send(Message::Binary(encoded_data)).await.unwrap();
                 }

--- a/src/test_web_socket.rs
+++ b/src/test_web_socket.rs
@@ -1,3 +1,4 @@
+use crate::WsMessage;
 use anyhow::anyhow;
 use anyhow::Context;
 use anyhow::Result;
@@ -13,11 +14,8 @@ use std::fmt::Display;
 use tokio_tungstenite::tungstenite::protocol::Role;
 use tokio_tungstenite::WebSocketStream;
 
-use crate::WsMessage;
-
 #[cfg(feature = "pretty-assertions")]
 use pretty_assertions::assert_eq;
-use tokio_tungstenite::tungstenite::Utf8Bytes;
 
 pub struct TestWebSocket {
     stream: WebSocketStream<TokioIo<Upgraded>>,
@@ -43,8 +41,7 @@ impl TestWebSocket {
         T: Display,
     {
         let text = format!("{}", raw_text);
-        let text_bytes: Utf8Bytes = text.try_into().expect("Text contains non-Utf8 chars");
-        self.send_message(WsMessage::Text(text_bytes)).await;
+        self.send_message(WsMessage::Text(text.into())).await;
     }
 
     pub async fn send_json<J>(&mut self, body: &J)
@@ -53,9 +50,8 @@ impl TestWebSocket {
     {
         let raw_json =
             ::serde_json::to_string(body).expect("It should serialize the content into Json");
-        let json_bytes: Utf8Bytes = raw_json.try_into().expect("Json contains non-Utf8 chars");
 
-        self.send_message(WsMessage::Text(json_bytes)).await;
+        self.send_message(WsMessage::Text(raw_json.into())).await;
     }
 
     #[cfg(feature = "yaml")]
@@ -65,9 +61,8 @@ impl TestWebSocket {
     {
         let raw_yaml =
             ::serde_yaml::to_string(body).expect("It should serialize the content into Yaml");
-        let yaml_bytes: Utf8Bytes = raw_yaml.try_into().expect("Yaml contains non-Utf8 chars");
 
-        self.send_message(WsMessage::Text(yaml_bytes)).await;
+        self.send_message(WsMessage::Text(raw_yaml.into())).await;
     }
 
     #[cfg(feature = "msgpack")]
@@ -233,9 +228,9 @@ fn message_to_text(message: WsMessage) -> Result<String> {
 fn message_to_bytes(message: WsMessage) -> Result<Bytes> {
     let bytes = match message {
         WsMessage::Text(string) => string.into(),
-        WsMessage::Binary(data) => data.into(),
-        WsMessage::Ping(data) => data.into(),
-        WsMessage::Pong(data) => data.into(),
+        WsMessage::Binary(data) => data,
+        WsMessage::Ping(data) => data,
+        WsMessage::Pong(data) => data,
         WsMessage::Close(None) => Bytes::new(),
         WsMessage::Close(Some(frame)) => frame.reason.into(),
         WsMessage::Frame(_) => {

--- a/src/transport_layer/into_transport_layer.rs
+++ b/src/transport_layer/into_transport_layer.rs
@@ -11,10 +11,11 @@ mod router;
 mod serve;
 mod with_graceful_shutdown;
 
-#[cfg(feature = "shuttle")]
-mod axum_service;
-#[cfg(feature = "shuttle")]
-mod shuttle_axum;
+// TODO, bring this back when Shuttle updates to Axum 8
+// #[cfg(feature = "shuttle")]
+// mod axum_service;
+// #[cfg(feature = "shuttle")]
+// mod shuttle_axum;
 
 ///
 /// This exists to unify how to send mock or real messages to different services.

--- a/src/transport_layer/into_transport_layer/into_make_service_with_connect_info.rs
+++ b/src/transport_layer/into_transport_layer/into_make_service_with_connect_info.rs
@@ -1,3 +1,8 @@
+use crate::internals::HttpTransportLayer;
+use crate::transport_layer::IntoTransportLayer;
+use crate::transport_layer::TransportLayer;
+use crate::transport_layer::TransportLayerBuilder;
+use crate::util::spawn_serve;
 use anyhow::anyhow;
 use anyhow::Result;
 use axum::extract::connect_info::IntoMakeServiceWithConnectInfo;
@@ -5,18 +10,13 @@ use axum::extract::Request as AxumRequest;
 use axum::response::Response as AxumResponse;
 use axum::serve::IncomingStream;
 use std::convert::Infallible;
+use tokio::net::TcpListener;
 use tower::Service;
 use url::Url;
 
-use crate::internals::HttpTransportLayer;
-use crate::transport_layer::IntoTransportLayer;
-use crate::transport_layer::TransportLayer;
-use crate::transport_layer::TransportLayerBuilder;
-use crate::util::spawn_serve;
-
 impl<S, C> IntoTransportLayer for IntoMakeServiceWithConnectInfo<S, C>
 where
-    for<'a> C: axum::extract::connect_info::Connected<IncomingStream<'a>>,
+    for<'a> C: axum::extract::connect_info::Connected<IncomingStream<'a, TcpListener>>,
     S: Service<AxumRequest, Response = AxumResponse, Error = Infallible> + Clone + Send + 'static,
     S::Future: Send,
 {

--- a/src/transport_layer/into_transport_layer/serve.rs
+++ b/src/transport_layer/into_transport_layer/serve.rs
@@ -1,3 +1,8 @@
+use crate::internals::HttpTransportLayer;
+use crate::transport_layer::IntoTransportLayer;
+use crate::transport_layer::TransportLayer;
+use crate::transport_layer::TransportLayerBuilder;
+use crate::util::ServeHandle;
 use anyhow::anyhow;
 use anyhow::Context;
 use anyhow::Result;
@@ -6,20 +11,17 @@ use axum::response::Response;
 use axum::serve::IncomingStream;
 use axum::serve::Serve;
 use std::convert::Infallible;
+use tokio::net::TcpListener;
 use tokio::spawn;
 use tower::Service;
 use url::Url;
 
-use crate::internals::HttpTransportLayer;
-use crate::transport_layer::IntoTransportLayer;
-use crate::transport_layer::TransportLayer;
-use crate::transport_layer::TransportLayerBuilder;
-use crate::util::ServeHandle;
-
-impl<M, S> IntoTransportLayer for Serve<M, S>
+impl<M, S> IntoTransportLayer for Serve<TcpListener, M, S>
 where
-    M: for<'a> Service<IncomingStream<'a>, Error = Infallible, Response = S> + Send + 'static,
-    for<'a> <M as Service<IncomingStream<'a>>>::Future: Send,
+    M: for<'a> Service<IncomingStream<'a, TcpListener>, Error = Infallible, Response = S>
+        + Send
+        + 'static,
+    for<'a> <M as Service<IncomingStream<'a, TcpListener>>>::Future: Send,
     S: Service<Request, Response = Response, Error = Infallible> + Clone + Send + 'static,
     S::Future: Send,
 {

--- a/src/transport_layer/into_transport_layer/with_graceful_shutdown.rs
+++ b/src/transport_layer/into_transport_layer/with_graceful_shutdown.rs
@@ -1,3 +1,8 @@
+use crate::internals::HttpTransportLayer;
+use crate::transport_layer::IntoTransportLayer;
+use crate::transport_layer::TransportLayer;
+use crate::transport_layer::TransportLayerBuilder;
+use crate::util::ServeHandle;
 use anyhow::anyhow;
 use anyhow::Context;
 use anyhow::Result;
@@ -6,21 +11,18 @@ use axum::response::Response;
 use axum::serve::IncomingStream;
 use axum::serve::WithGracefulShutdown;
 use std::convert::Infallible;
+use std::future::Future;
+use tokio::net::TcpListener;
 use tokio::spawn;
 use tower::Service;
 use url::Url;
 
-use crate::internals::HttpTransportLayer;
-use crate::transport_layer::IntoTransportLayer;
-use crate::transport_layer::TransportLayer;
-use crate::transport_layer::TransportLayerBuilder;
-use crate::util::ServeHandle;
-use std::future::Future;
-
-impl<M, S, F> IntoTransportLayer for WithGracefulShutdown<M, S, F>
+impl<M, S, F> IntoTransportLayer for WithGracefulShutdown<TcpListener, M, S, F>
 where
-    M: for<'a> Service<IncomingStream<'a>, Error = Infallible, Response = S> + Send + 'static,
-    for<'a> <M as Service<IncomingStream<'a>>>::Future: Send,
+    M: for<'a> Service<IncomingStream<'a, TcpListener>, Error = Infallible, Response = S>
+        + Send
+        + 'static,
+    for<'a> <M as Service<IncomingStream<'a, TcpListener>>>::Future: Send,
     S: Service<Request, Response = Response, Error = Infallible> + Clone + Send + 'static,
     S::Future: Send,
     F: Future<Output = ()> + Send + 'static,

--- a/src/util/spawn_serve.rs
+++ b/src/util/spawn_serve.rs
@@ -1,23 +1,25 @@
+use crate::util::ServeHandle;
 use axum::extract::Request;
 use axum::response::Response;
 use axum::serve;
 use axum::serve::IncomingStream;
+use axum::serve::Listener;
+use core::fmt::Debug;
 use std::convert::Infallible;
-use tokio::net::TcpListener;
 use tokio::spawn;
 use tower::Service;
-
-use crate::util::ServeHandle;
 
 /// A wrapper around [`axum::serve()`] for tests,
 /// which spawns the service in a new thread.
 ///
 /// The [`crate::util::ServeHandle`] returned will automatically attempt
 /// to terminate the service when dropped.
-pub fn spawn_serve<M, S>(tcp_listener: TcpListener, make_service: M) -> ServeHandle
+pub fn spawn_serve<L, M, S>(tcp_listener: L, make_service: M) -> ServeHandle
 where
-    M: for<'a> Service<IncomingStream<'a>, Error = Infallible, Response = S> + Send + 'static,
-    for<'a> <M as Service<IncomingStream<'a>>>::Future: Send,
+    L: Listener,
+    L::Addr: Debug,
+    M: for<'a> Service<IncomingStream<'a, L>, Error = Infallible, Response = S> + Send + 'static,
+    for<'a> <M as Service<IncomingStream<'a, L>>>::Future: Send,
     S: Service<Request, Response = Response, Error = Infallible> + Clone + Send + 'static,
     S::Future: Send,
 {

--- a/test.sh
+++ b/test.sh
@@ -3,7 +3,7 @@
 set -e
 
 cargo check
-cargo test --example=example-shuttle --features shuttle
+# cargo test --example=example-shuttle --features shuttle
 cargo test --example=example-todo
 cargo test --example=example-websocket-ping-pong --features ws
 cargo test --example=example-websocket-chat --features ws
@@ -17,7 +17,7 @@ cargo check --features pretty-assertions
 cargo check --features yaml
 cargo check --features msgpack
 cargo check --features reqwest
-cargo check --features shuttle
+# cargo check --features shuttle
 cargo check --features typed-routing
 cargo check --features ws
 cargo check --features reqwest

